include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -96,4 +97,56 @@ def XTenNN_OutputOp : XTenNN_Op<"output", [
 }
 
 
+def XTenNN_QuantizeOp: XTenNN_Op<"quantize", [
+    SameOperandsAndResultShape]> {
+  
+
+  let summary = "quantizes a float32 tensor to a signed or unsigned integer tensor of given width.";
+
+  let description = [{
+    Quantizes a given float32 tensor into a signed or unsigned integer tensor of given width.
+
+    Applies the following to linear quantization the input tensor x:
+      y = round( x * scale ) + zero_point
+
+      Where scale is an attribute of the operation in f32 and is a power-of-two value.
+      While the zero_point is always assumed to be zero.
```

Comment:
It's a little odd to have a zero_point in the formula but then says it's not there

Reply:
That is true, but I wanted to keep the operation as simple as possible well still documenting where it is coming from. I think I will add a zero_point but constrain the value.